source/CardGames.py

```diff
@@ -104,6 +104,17 @@ def clearHand(self):
     self.hand = []
     self.knownCards = []
 
+  def getPairs(self):
+    pairList = []
+    for x in self.hand:
+      for y in self.hand:
+        if x.value == y.value:
+          if x.__eq__(y)== False and ([y,x] in pairList) == False:
+            pairList.append([x,y])
+
+
+    return pairList
+
```

Comment:
Nice Job. Only one small thing I would add is comments. I don't fully understand why you have the last "[y,x'] in pairList". I'm sure you have a good reason. Other than that, excellent work.